examples/all-clusters-app/linux/DRLC.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/clusters/drlc-server/drlc-server.h>
+#include <lib/support/CodeUtils.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::DataModel;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::DemandResponseLoadControl;
+using Status = Protocols::InteractionModel::Status;
+
+class OurDelegate : public DemandResponseLoadControl::Delegate
+{
+public:
+    uint8_t GetNumberOfLoadControlPrograms() override;
+    uint8_t GetNumberOfEventsPerProgram() override;
+    uint8_t GetNumberOfTransitions() override;
+    uint8_t GetDefaultRandomStart() override;
+    CHIP_ERROR SetDefaultRandomStart(uint8_t aNewValue) override;
+    uint8_t GetDefaultRandomDuration() override;
+    CHIP_ERROR SetDefaultRandomDuration(uint8_t aNewValue) override;
+    CHIP_ERROR GetLoadControlProgram(size_t aIndex, LoadControlProgram & aLoadControlProgram) override;
+    CHIP_ERROR GetActiveEventID(size_t aIndex, FixedSpan<uint8_t, kEventIDSize> & aEventID) override;
+    Status ReplaceLoadControlProgram(size_t aIndex, const Structs::LoadControlProgramStruct::DecodableType & aNewProgram) override;
+    Status AddLoadControlProgram(const Structs::LoadControlProgramStruct::DecodableType & aNewProgram) override;
+    Status RemoveLoadControlProgram(const ByteSpan & aProgramID) override;
+    Status AddLoadControlEvent(const Structs::LoadControlEventStruct::DecodableType & aEvent) override;
+    Status ClearLoadControlEvents() override;
```

Comment:
In Hasty's latest spec PR: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7762 
these have been changed to
| 0x00 s| RegisterLoadControlProgramRequest   | client => server | Y        | O      | M
| 0x01 s| UnregisterLoadControlProgramRequest | client => server | Y        | O      | M
| 0x02 s| AddLoadControlEventRequest          | client => server | Y        | O      | M
| 0x03 s| RemoveLoadControlEventRequest       | client => server | Y        | O      | M

Reply:
That PR keeps changing.   :(  Makes it very hard to target.  Will update.